Client/src/Unified/Services/Submitter/Service.cs

```diff
@@ -284,194 +277,141 @@ public string SessionId
                           20000,
                           30000,
                         };
-    var idx = 0;
+    var emptyFetch = 0;
 
-    while (missing.Count != 0)
+    while (!(CancellationResultTaskSource.Token.IsCancellationRequested && ResultHandlerDictionary.IsEmpty))
     {
-      foreach (var bucket in missing.ToList()
-                                    .ToChunk(chunkResultSize))
+      try
       {
-        var resultStatusCollection = SessionService.GetResultStatus(bucket);
+        if (ResultHandlerDictionary.IsEmpty)
+        {
+          Thread.Sleep(100);
+          continue;
+        }
+
+        var fetched = 0;
 
-        foreach (var resultStatusData in resultStatusCollection.IdsReady)
+        foreach (var chunk in ResultHandlerDictionary.Keys.ToChunk(500))
         {
-          try
-          {
-            Logger?.LogTrace("Response handler for {taskId}",
-                             resultStatusData.TaskId);
-            responseHandler(resultStatusData.TaskId,
-                            SessionService.TryGetResultAsync(new ResultRequest
-                                                             {
-                                                               ResultId = resultStatusData.ResultId,
-                                                               Session  = SessionId,
-                                                             },
-                                                             CancellationToken.None)
-                                          .Result);
-          }
-          catch (Exception e)
+          var resultStatusCollection = SessionService.GetResultStatus(chunk);
+
+          foreach (var resultStatusData in resultStatusCollection.IdsReady)
           {
-            Logger?.LogWarning(e,
-                               "Response handler for {taskId} threw an error",
-                               resultStatusData.TaskId);
+            var taskId = resultStatusData.TaskId;
+            fetched += 1;
+
             try
             {
-              errorHandler(resultStatusData.TaskId,
-                           TaskStatus.Error,
-                           e.Message + e.StackTrace);
+              var byteResult = SessionService.TryGetResultAsync(new ResultRequest
+                                                                {
+                                                                  ResultId = resultStatusData.ResultId,
+                                                                  Session  = SessionId,
+                                                                },
+                                                                CancellationToken.None)
```

Comment:
No usage of CancellationResultTaskSource.Token instead of CancellationToker.None ?